fbw-a32nx/src/systems/extras-host/modules/baro_handler/BaroHandler.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2023 FlyByWire Simulations
+// SPDX-License-Identifier: GPL-3.0
+
+import { EventBus, KeyEvents, KeyEventData, KeyInterceptManager } from 'msfssdk';
+import { Arinc429Register } from '@shared/arinc429';
+import { FmgcFlightPhase } from '@shared/flightphase';
+
+export class BaroHandler {
+    private readonly fm1TransAlt = Arinc429Register.empty();
+
+    private readonly fm2TransAlt = Arinc429Register.empty();
+
+    private readonly altitude = Arinc429Register.empty();
+
+    constructor(private readonly bus: EventBus) {}
+
+    public connectedCallback(): void {
+        this.bus.getSubscriber<KeyEvents>().on('key_intercept').handle(this.handleKey.bind(this));
+
+        KeyInterceptManager.getManager(this.bus).then((manager) => {
+            manager.interceptKey('BAROMETRIC', false);
+        });
+    }
+
+    private async handleKey(ev: KeyEventData): Promise<void> {
+        if (ev.key === 'BAROMETRIC') {
+            // FIXME handle F/O side separately in future when it's split off
+
+            const fmFlightPhase = SimVar.GetSimVarValue('L:A32NX_FMGC_FLIGHT_PHASE', 'number');
+            const climb = fmFlightPhase <= FmgcFlightPhase.Climb;
+
+            // try all the ADRs until we find a healthy one
+            this.altitude.set(0);
+            for (let n = 1; n <= 3; n++) {
+                if (climb && !this.altitude.isNormalOperation()) {
+                    // use baro correct alt as we're climbing through the transition alt
+                    this.altitude.setFromSimVar(`L:A32NX_ADIRS_ADR_${n}_BARO_CORRECTED_ALTITUDE_1`);
+                }
+                if (!climb && !this.altitude.isNormalOperation()) {
+                    // use pressure alt as we're descending through the transition level
+                    this.altitude.setFromSimVar(`L:A32NX_ADIRS_ADR_${n}_ALTITUDE`);
+                }
+            }
+
+            if (!this.altitude.isNormalOperation()) {
+                // we can't do anything as there are no healthy ADRs
```

Comment:
Might warrant a console output (log, warning) - not fully clear to me if this is a kind of error condition or not.

Reply:
Just means all the ADRs are off or failed. Not a code error.